app/controllers/checkouts_controller.rb

```diff
@@ -0,0 +1,40 @@
+class CheckoutsController < Customers::BaseController
+  include MessageHelper
+  before_action :checkout, only: [:new]
+
+  def new
+  end
+
+  def create
+    customer = current_customer.stripe_customer
+
+    begin
```

Comment:
there's a lot of stuff going on here (i know this is all stripe logic) - i suggest pulling this out to a PORO in order to keep your controller more single responsibility 👍